pkg/multicluster/link.go

```diff
@@ -368,7 +388,7 @@ func newProbeSpec(obj map[string]interface{}) (ProbeSpec, error) {
 func stringField(obj map[string]interface{}, key string) (string, error) {
 	value, ok := obj[key]
 	if !ok {
-		return "", fmt.Errorf("Field '%s' is missing", key)
+		return "", &ErrFieldMissing{Field: key}
```

Comment:
TIOLI, I find it idiomatic and ergonomic to return a zero value to represent "not found" instead of an error.  i.e. `return "", nil`

Reply:
That makes sense, but then the calling side would have to have logic for distinguishing between the two errors (field missing vs field not a string) which is what this function is all about. I see my change kinda does that for `failureThreshold` and `timeout`, but not for the other fields getting validated with this function.